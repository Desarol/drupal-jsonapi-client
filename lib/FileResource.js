"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _Resource = _interopRequireDefault(require("./Resource"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class FileResource extends _Resource.default {
  static New({
    type,
    field,
    file,
    fileName
  }) {
    const [p1, p2] = type.split('--');
    return {
      url: `/${p1}/${p2}/${field}`,
      method: 'POST',
      headers: {
        Accept: 'application/vnd.api+json',
        'Content-Type': 'application/octet-stream',
        'Content-Disposition': `file; filename="${fileName}"`
      },
      data: file
    };
  }

}

var _default = FileResource;
exports.default = _default;