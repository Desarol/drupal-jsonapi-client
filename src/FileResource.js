import Resource from './Resource'

class FileResource extends Resource {
  static New({
    type,
    field,
    file,
    fileName,
  }) {
    const [p1, p2] = type.split('--')

    return {
      url: `/${p1}/${p2}/${field}`,
      method: 'POST',
      headers: {
        Accept: 'application/vnd.api+json',
        'Content-Type': 'application/octet-stream',
        'Content-Disposition': `file; filename="${fileName}"`,
      },
      data: file,
    }
  }
}

export default FileResource
