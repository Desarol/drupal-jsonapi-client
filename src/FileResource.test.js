import FileResource from './FileResource'

describe('FileResource', () => {
  it('generates a valid request from a .New call', () => {
    const request = FileResource.New({
      type: 'node--article',
      field: 'field_image',
      file: new Uint8Array(),
      fileName: 'file.jpg',
    })

    expect(request).toMatchSnapshot()
  })
})
