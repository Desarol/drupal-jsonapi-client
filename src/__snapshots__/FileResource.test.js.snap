// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileResource generates a valid request from a .New call 1`] = `
Object {
  "data": Uint8Array [],
  "headers": Object {
    "Accept": "application/vnd.api+json",
    "Content-Disposition": "file; filename=\\"file.jpg\\"",
    "Content-Type": "application/octet-stream",
  },
  "method": "POST",
  "url": "/node/article/field_image",
}
`;
